clusterloader2/pkg/config/codec.go

```diff
@@ -58,3 +64,25 @@ func ReadTemplate(path string) (*unstructured.Unstructured, error) {
 	}
 	return obj, nil
 }
+
+// ReadTemplateWithCache creates object template from file specified by the given path
+// or uses cached object if available.
+func ReadTemplateWithCache(path string) (*unstructured.Unstructured, error) {
+	lock.RLock()
```

Comment:
Guess you can just lock and defer unlock once at the top of the function to make is simple (as you're accessing multiple times).

Reply:
RLock is for read-only and can be acquire be multiple threads at the same time - will be used very often.
Lock blocks all the access to object - will be used only for initialization.

I want to separate this two cases and get better performance due to RLock being used.

Reply:
Yes, that sounds quite reasonable with read-heavy load, which is the case here.